integration/pubsub_test.go

```diff
@@ -401,13 +471,26 @@ func checkResultHandler(publishedMsg string, publishedMsgType string, ch chan<-
 				return message.Error(errors.New("published message type and received message type are different"))
 			}
 			if publishedMsgType == reflection.GetType(retryLaterEvent{}) {
-				//use delivery count now that retry later abandons
+				// use delivery count now that retry later abandons
 				if msg.Message().DeliveryCount == 2 {
 					ch <- true
 					return message.Complete()
 				}
 				return message.RetryLater(1 * time.Second)
 			}
+			if publishedMsgType == reflection.GetType(shortLockMessage{}) {
+				// the shortlock is set at 1 second in the listener setup
+				// We sleep for longer to trigger the lock renewal in the listener.
+				// if the renewal works, then the message completion will succeed.
+				time.Sleep(3 * time.Second)
+				res := message.Complete()
```

Comment:
if renewal wasn't working we'd fail here or would renewal cancel this? Worth an integration test to show we fail if we lost the lock?

Reply:
yes, it will fail Completing the message.
Once we have a per message context, we can cancel that context.
the behavior of the message handling will be the same though, since canceling isn't stopping processing, it's just preventing future calls that use the context.
The improvement will be that we'll fail on our side rather than after the RPC call.
so, leaving it as is until we have per message contexts.